docs/extras/modules/data_connection/vectorstores/integrations/googlevertexai.mdx

```diff
@@ -0,0 +1,176 @@
+---
+sidebar_class_name: node-only
+---
+
+import CodeBlock from "@theme/CodeBlock";
+
+# Google Vertex AI Matching Engine
+
+:::tip Compatibility
+Only available on Node.js.
+:::
+
+The Google Vertex AI Matching Engine "provides the industry's leading high-scale
+low latency vector database. These vector databases are commonly referred
+to as vector similarity-mathing or an aprximate nearest neighbor (ANN) service."
+
+## Setup
+
+:::caution
+This module expects an endpoint and deployed index already created as the
+creation time takes close to one hour. To learn more, see the LangChain python
+documentation [Create Index and deploy it to an Endpoint](https://python.langchain.com/docs/integrations/vectorstores/matchingengine#create-index-and-deploy-it-to-an-endpoint).
+:::
+
+Before running this code, you should make sure the Vertex AI API is
+enabled for the relevant project in your Google Cloud dashboard and that you've authenticated to
+Google Cloud using one of these methods:
+
+- You are logged into an account (using `gcloud auth application-default login`)
+  permitted to that project.
+- You are running on a machine using a service account that is permitted
+  to the project.
+- You have downloaded the credentials for a service account that is permitted
+  to the project and set the `GOOGLE_APPLICATION_CREDENTIALS` environment
+  variable to the path of this file.
+
+Install the authentication library with:
+
+```bash npm2yarn
+npm install google-auth-library
+```
+
+## Usage
+
+### Initializing the engine
+
+When creating the `MatchingEngine` object, you'll need some information about
+the matching engine configuration. You can get this information from the Cloud Console
+for Matching Engine:
+
+- The id for the Index
+- The id for the Index Endpoint
+
+You will also need a document store. While an `InMemoryDocstore` is ok for
+initial testing, you will likely want to use something like a
+`GoogleCloudStorageDocstore` to store it more permanently.
+
+```typescript
+    import { MatchingEngine } from "langchain/vectorstores/googlevertexai";
```

Comment:
Add under `/examples` and import it instead?

Reply:
I'll add the whole thing under /examples, but I think doing it with imports would do one of the following:

- If I have it as a single example, I can't easily break it down into the smaller, explainable, parts.
- If I have multiple examples that get imported, then there is a lot of repeated code (setting up the Matching Engine class) and the relevant example may get lost.

I know best practice is to import the example, but I think that will make the documentation less useful.

Thoughts?

Reply:
Yeah no worries then - just make sure it runs as is!

We need a more Jupyter notebook style of solution long-term.

Reply:
Yup. `yarn example indexes/vector_stores/googlevertexai` runs with the environment variables set. Tested and confirmed that.